test/request.js

```diff
@@ -34,46 +34,58 @@ describe('#unit Check request response error callback', function() {
       .get('/good')
       .reply(200, { ok: true });
 
-    db.get('good', function(err) {
-      err = error.convertResponseError(err);
-      assert.strictEqual(err, undefined);
+    db.service.getDocument({ db: db.db, docId: 'good' }).then(response => {
+      assert.ok(response.result);
       assert.ok(couch.isDone());
       done();
+    }).catch(err => {
+      err = error.convertResponseError(err);
+      done(err);
     });
   });
 
   it('should callback with error after 3 500 responses', function(done) {
     var couch = nock(url)
       .get('/bad')
       .times(3)
-      .reply(500, { error: 'foo', reason: 'bar' });
+      .reply(500, function(uri, requestBody) {
+        this.req.response.statusMessage = 'Internal Server Error';
+        return { error: 'foo', reason: 'bar' };
+      });
 
-    db.get('bad', function(err) {
+    db.service.getDocument({ db: db.db, docId: 'bad' }).then(response => {
+      done(new Error('Successful response when error expected.'));
+    }).catch(err => {
       err = error.convertResponseError(err);
       assert.strictEqual(err.name, 'HTTPFatalError');
-      assert.strictEqual(err.message, `500 : GET ${url}/bad - Error: foo, Reason: bar`);
+      assert.strictEqual(err.message, '500 foo - Error: foo, Reason: bar');
```

Comment:
Are the tests checking somewhere there were 3 attempts?

Reply:
yes, the `couch.isDone()` on the next line validates that all the queued responses were returned (i.e. `times(3)` was satisfied in this case).
If a fourth request happens nock will throw an error that there was no matching request.